### PR TITLE
Fixed to be able to use env of Step

### PR DESCRIPTION
**Pull request by hnarimiya**
###  Summary
Hello, thanks for this useful software !

I modified the env defined in step to be available in file.

Also, the process of replacing $ was strange, so I fixed it.

In addition, notes on using files were added to the README.

issues: https://github.com/slackapi/slack-github-action/issues/84

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/slack-github-action/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

**Comment by WilliamBergamin** [comment]
Hi @hnarimiya thanks for the contribution 💯 

I'm currently working on a work around for these integration test, will wait on others approval before merging this

**Comment by satellitevu-dale** [comment]
What's the status of this? Would be good to get this fixed merged :)

File: .github/resources/payload-notification.json
Comment (filmaj) [inline .github/resources/payload-notification.json:1]: Some of the fields in this payload are not showing up correctly. Attached is a screenshot of the GitHub CI run from https://github.com/slackapi/slack-github-action/pull/163 (which is a copy of this PR), and how it looks like in our Slack workspace where the message gets posted:

Screenshot 2023-02-28 at 3 32 26 PM

In particular, it seems the following fields do not work as desired from the `integration_test_file_payload` test from `main.yml`: `env.JOB_STATUS`, `github.payload.head_commit.url`, `github.payload.head_commit.author.name`, `github.payload.head_commit.message`.

Any idea why this might be happening?

**Review by filmaj — CHANGES_REQUESTED** [review]
It doesn't look like these changes render properly, at least based on the integration tests we ran with these changes. Any idea why?

File: .github/resources/payload-notification.json
Comment (rd-florian-stagliano) [reply .github/resources/payload-notification.json:1]: same seems to happen for us

**Comment by filmaj** [comment]
We integrated this fix in #200, will release shortly.

**Comment by Darwiner** [comment]
Using v1.24.0, I just tried switching to using `payload-file-path:` and it seems to still not be working properly.

The rendering works fine using `payload:`.

Anyone else?

image

**Comment by Darwiner** [comment]
> @Darwiner what event tripped the action and what does the file look like? We found in our testing that different events have different shapes and so those details matter.

This was triggered via a PR open/synchronize, the top of the workflow file being:
```
on:
  workflow_dispatch:
  pull_request:
    types:
      - opened
      - synchronize
    branches:
      - main
      - master
```

As for the json file, it goes like this:
```
{
  "text": "GitHub Actions - Slack - Deploy Success",
  "blocks": [
    {
      "type": "section",
      "text": {
        "type": "mrkdwn",
        "text": ":rocket: *DEPLOY* - <${{ github.event.repository.html_url }}|${{ steps.vars.outputs.app }}>"
      }
    },
    {
      "type": "divider"
    },
    {
      "type": "section",
      "fields": [
        {
          "type": "mrkdwn",
          "text": "*Cluster* - ${{ steps.vars.outputs.gcp-project }} / ${{ inputs.cluster-location }} / ${{ inputs.cluster-name }}"
        },
        {
          "type": "mrkdwn",
          "text": "*Env* - ${{ steps.vars.outputs.env }}"
        },
        {
          "type": "mrkdwn",
          "text": "*Release* - ${{ steps.vars.outputs.release-version }}"
        },
        {
          "type": "mrkdwn",
          "text": "*PR* - <${{ github.event.pull_request.html_url || github.event.head_commit.url }}|${{ github.event.pull_request.title || github.event.head_commit.message }}>"
        },
        {
          "type": "mrkdwn",
          "text": "*Workflow* - <${{ github.event.repository.html_url }}/actions/runs/${{ github.run_id }}|Logs>"
        }
      ]
    },
    {
      "type": "divider"
    },
    {
      "type": "section",
      "text": {
        "type": "mrkdwn",
        "text": "*STATUS* - ${{ github.action_status }} :white_check_mark:"
      }
    }
  ]
}
```

Additionally, another piece of info that might or might not be relevant is that this is triggered via a composite action.

The action in the composite action being:

```
    - id: slack-prod-success
      uses: slackapi/slack-github-action@v1.24.0
      with:
        channel-id: ${{ inputs.slack-success-channel }}
        payload-file-path: ${{ github.action_path }}/payload-slack-success.json
      env:
        SLACK_BOT_TOKEN: ${{ inputs.slack-bot-token }}
```

**Comment by rd-florian-stagliano** [comment]
we have the same problem. the 'fix' didn't the rendering when you use a payload json file. We also see the github env variables rendered with '?'

**Comment by filmaj** [comment]
I've filed #203 to follow-up more deeply.

We use this Action in this action's intergration tests, and they render correctly. So something weird is going on.

Will investigate more in #203.

**Pull request closed without merging**
